docs/framework/winforms/controls/how-to-make-thread-safe-calls-to-windows-forms-controls.md

```diff
@@ -17,1032 +16,309 @@ helpviewer_keywords:
-
-3.  Bind the method created in step 1 to the <xref:System.ComponentModel.BackgroundWorker.DoWork> event of an instance of <xref:System.ComponentModel.BackgroundWorker>, and bind the method created in step 2 to the same instance’s <xref:System.ComponentModel.BackgroundWorker.RunWorkerCompleted> event.
-
-4.  To start the background thread, call the <xref:System.ComponentModel.BackgroundWorker.RunWorkerAsync%2A> method of the <xref:System.ComponentModel.BackgroundWorker> instance.
-
-In the following code example, the <xref:System.ComponentModel.BackgroundWorker.DoWork> event handler uses <xref:System.Threading.Thread.Sleep%2A> to simulate work that takes some time. It does not call the form’s <xref:System.Windows.Forms.TextBox> control. The <xref:System.Windows.Forms.TextBox> control's <xref:System.Windows.Forms.Control.Text%2A> property is set directly in the <xref:System.ComponentModel.BackgroundWorker.RunWorkerCompleted> event handler.
-
-```csharp
-// This BackgroundWorker is used to demonstrate the
-// preferred way of performing asynchronous operations.
-private BackgroundWorker backgroundWorker1;
+    private void SetText()
+    {
+        SafeText("This text was set safely.");
+    }
+    private void Button1_Click(object sender, EventArgs e)
+    {
+        thread2 = new Thread(new ThreadStart(SetText));
+        thread2.Start();
+        Thread.Sleep(1000);
+    }
+}
 ```
 
 ```vb
-' This BackgroundWorker is used to demonstrate the
-' preferred way of performing asynchronous operations.
-Private WithEvents backgroundWorker1 As BackgroundWorker
-```
+Imports System.Drawing
+Imports System.Threading
+Imports System.Windows.Forms
 
-```cpp
-// This BackgroundWorker is used to demonstrate the
-    // preferred way of performing asynchronous operations.
-private:
-    BackgroundWorker^ backgroundWorker1;
-```
+Public Class InvokeThreadSafeForm : Inherits Form
 
-```csharp
-// This event handler starts the form's
-// BackgroundWorker by calling RunWorkerAsync.
-//
-// The Text property of the TextBox control is set
-// when the BackgroundWorker raises the RunWorkerCompleted
-// event.
-private void setTextBackgroundWorkerBtn_Click(
-	object sender,
-	EventArgs e)
-{
-	this.backgroundWorker1.RunWorkerAsync();
-}
+    Public Shared Sub Main()
+        Application.SetCompatibleTextRenderingDefault(False)
+        Application.EnableVisualStyles()
+        Dim frm As New InvokeThreadSafeForm()
+        Application.Run(frm)
+    End Sub
 
-// This event handler sets the Text property of the TextBox
-// control. It is called on the thread that created the
-// TextBox control, so the call is thread-safe.
-//
-// BackgroundWorker is the preferred way to perform asynchronous
-// operations.
+    Dim WithEvents Button1 As Button
+    Dim TextBox1 As TextBox
+    Dim Thread2 as Thread = Nothing
+
+    Delegate Sub SafeCallDelegate([text] As String)
```

Comment:
I don't think that brackets around `text` are necessary; make sure the example compiles without them and, if it does, remove them here and elsewhere.

Reply:
@rpetrusha - It does compile without the brackets and I have removed them.